### ISSUE TITLE
Common eslint and stylelint configs for official Polymer elements

### ISSUE THREAD

**Issue by web-padawan** [Polymer/project#35]
In the last few days I had to send two PRs just to fix wrong semicolons in [iron-list](https://github.com/PolymerElements/iron-list/issues/387) and [paper-input](https://github.com/PolymerElements/paper-input/issues/475).

Previously there was such an issue with [iron-flex-layout](https://github.com/PolymerElements/iron-flex-layout/issues/97), which is already fixed.

Sometimes such small issues can occasionally break tools like `clean-css` or `uglifyjs`, so it's very sad that you guys don't have a `stylelint` and `eslint` config.

I created  [stylelint-config-polymer](https://github.com/web-padawan/stylelint-config-polymer) with some options for `:host` and `::content` selectors, and regexp allowing Custom elements (tag names with dashes).
I would be happy if you guys decide to make it official (and I would transfer ownership of course).

This means mass travis config changes (and maybe some other issues will be discovered), but I hope that `tedium` can deal with it, while [tattoo](https://github.com/Polymer/tattoo) could also implement an option to lint all the things.

/cc @robdodson @notwaldorf

**Comment by abdonrd** [comment]
I like the idea! Proposal to ESLint:

```json
{
  "extends": "eslint:recommended",
  "env": {
    "browser": true
  },
  "plugins": [
    "html"
  ],
  "settings": {
    "html/indent": "+2",
    "html/report-bad-indent": 2
  },
  "globals": {
    "Polymer": true
  }
}
```

**Comment by web-padawan** [comment]
UPD: `stylelint` already supports `host`, `host-context` and `slotted` selectors, no need to separate config,we can just use `stylelint-config-standard`.

**Comment by web-padawan** [comment]
Looks like the only meed for now is to allow custom element selectors. 
I do it via RegEpx.  Here is my proposal to stylelint.
```
{
    "extends": "stylelint-config-standard",
    "processors": [ "stylelint-processor-html" ],
    "rules": {
        "selector-type-no-unknown": [ true, {
            "ignoreTypes": ["/^[a-zA-Z]([a-zA-Z0-9]*-[a-zA-Z0-9]+)+/"],
        } ]
    }
}
```

**Comment by abdonrd** [comment]
I propose add support to Custom Elements:

https://github.com/stylelint/stylelint/issues/2348

**Comment by abdonrd** [comment]
Stylelint suggest create a plugin meanwhile Custom Elements spec gains traction: https://github.com/stylelint/stylelint/issues/2348#issuecomment-279173136

**Comment by web-padawan** [comment]
I don't think that one rule is worth creating plugin, unless someone wants to use emoji in CE names. For alphanumeric characters, RegExp from the above snippet should work fine. 
After upcoming Safari 10.1 release, there will be two independent CE v1 implementations and I hope `stylelint` developers will re-consider implementing that.

**Comment by abdonrd** [comment]
Released in Stylelint v7.11.0!

https://github.com/stylelint/stylelint/releases/tag/7.11.0

**Comment by web-padawan** [comment]
@abdonrd 7.11.0 has a CSS mixins support regression reported [here](https://github.com/stylelint/stylelint/issues/2631).

**Comment by abdonrd** [comment]
@web-padawan oh, thanks!

**Comment by jeddy3** [comment]
@web-padawan Thanks for reporting the regression. It should be fixed in `stylelint@7.11.1`.

**Comment by stramel** [comment]
So I started on a config. I welcome any changes and suggestions :+1: 
https://github.com/stramel/eslint-plugin-polymer
